src/MongoDB.Driver.Core/Core/Operations/RetryabilityHelper.cs

```diff
@@ -86,17 +87,20 @@ static RetryabilityHelper()
         }
 
         // public static methods
-        public static void AddRetryableWriteErrorLabelIfRequired(MongoException exception)
+        public static void AddRetryableWriteErrorLabelIfRequired(MongoException exception, SemanticVersion serverVersion)
         {
-            if (ShouldRetryableWriteExceptionLabelBeAdded(exception))
+            if (ShouldRetryableWriteExceptionLabelBeAdded(exception, serverVersion))
             {
                 exception.AddErrorLabel(RetryableWriteErrorLabel);
             }
         }
 
-        public static bool IsNetworkException(Exception exception)
+        public static bool IsCommandRetryable(BsonDocument command)
         {
-            return exception is MongoConnectionException mongoConnectionException && mongoConnectionException.IsNetworkException;
+            return
+                command.Contains("txnNumber") || // retryWrites=true
```

Comment:
AFAIK a command is not retryable if it is in a transaction.

Do we need to check here whether the session is in a transaction or not?

Reply:
p.s. not sure how you would know here if we were in a transaction or not.

Reply:
>Do we need to check here whether the session is in a transaction or not?

I can add this validation, but it's not 100% required since we cannot have txtNumber is being filled if the operation within a transaction based on this logic: https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Operations/RetryableWriteOperationExecutor.cs#L149

Reply:
Let me know what do you think of whether this logic is required or not. If so, it will look similar to:

        public static bool IsCommandRetryable(BsonDocument command, bool isInTransaction)
        {
            return
                ((command.Contains("txnNumber") && !isInTransaction)) || // retryWrites=true
                command.Contains("commitTransaction") ||
                command.Contains("abortTransaction");
        }

Reply:
I think it's OK to assume that if the command contains "txnNumber" that we know that is not in a transaction.

Thanks for the clarification.